### PR TITLE
[CI-7416] - Update Save and Restore Cache from GCS topic

### PR DESCRIPTION
**Pull request by aimurphy**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [x] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* CI-7416: Add information about the GCP connector authentication requirements. Reorganize the page to make it easier to read. Remove an incorrect screenshot. Add an example of restore + save in the same stage.

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-integration/ci-technical-reference/restore-cache-from-gcs-settings.md [hunk 1]
```diff
@@ -32,9 +32,11 @@ GCS bucket name where you saved the cache.
 
 ## Key
 
-The key to identify the cache.
+The key identifying the cache to restore.
 
-The backslash character isn't allowed as part of the checksum value here. This is a limitation of the Go language (golang) template. You must use a forward slash instead:
+You can use the checksum macro to restore a key based on a file's checksum, for example: `myApp-{{ checksum filePath1 }}`
```
Comment (ravilach) [inline docs/continuous-integration/ci-technical-reference/restore-cache-from-gcs-settings.md:37]: Did we want to use the same "package.json" as the key here that is in the YAML later on?

**Review by ravilach — COMMENTED** [review]
Looks good to me. One comment left in line. Thanks for helping us make this better!

File: docs/continuous-integration/use-ci/caching-ci-data/save-cache-in-gcs.md [hunk 2]
```diff
@@ -1,49 +1,168 @@
...
+```
 
-The following YAML file defines a Pipeline with two Stages: Save Cache and Restore Cache. Copy the following YAML, paste it in the Harness YAML editor, and modify its attributes based on your Pipeline requirements.
+To configure the **Save Cache to GCS** step in the YAML editor, you must specify `type: SaveCacheGCS`. In `step: spec:`, specify the following:
+
+* `connectorRef` The GCP connector.
+* `bucket`: The GCS cache bucket.
+* `key`: The GCS cache key to identify the cache.
+* `sourcePaths`: Files and folders to cache. specify each file or folder separately.
+* `archiveFormat`: The archive format. The default format is `Tar`.
```
Comment (michaelcretzman) [inline docs/continuous-integration/use-ci/caching-ci-data/save-cache-in-gcs.md:83]:
```suggestion
* `connectorRef` The GCP connector Id.
* `bucket`: The GCS cache bucket name.
* `key`: The GCS cache key to identify the cache.
* `sourcePaths`: Files and folders to cache. Specify each file or folder separately.
* `archiveFormat`: The archive format. The default format is `Tar`.
```

File: docs/continuous-integration/use-ci/caching-ci-data/save-cache-in-gcs.md [hunk 3]
```diff
@@ -1,49 +1,168 @@
...
+                    bucket: ci_cache
+                    key: gcp-{{ checksum "package.json" }}
+                    sourcePaths:
+                      - /harness/node_modules
+                    archiveFormat: Tar
+```
 
+To configure the **Restore Cache from GCS** step in the YAML editor, you must specify `type: RestoreCacheGCS`. In `step: spec:`, specify the following:
+
+* `connectorRef` The GCP connector.
```
Comment (michaelcretzman) [inline docs/continuous-integration/use-ci/caching-ci-data/save-cache-in-gcs.md:101]:
```suggestion
* `connectorRef` The GCP connector Id.
```

File: docs/continuous-integration/use-ci/caching-ci-data/save-cache-in-gcs.md [hunk 4]
```diff
@@ -1,49 +1,168 @@
...
+* `bucket`: The GCS cache bucket. This must correspond with the Save Cache to GCS `bucket`.
```
Comment (michaelcretzman) [inline docs/continuous-integration/use-ci/caching-ci-data/save-cache-in-gcs.md:102]:
```suggestion
* `bucket`: The GCS cache bucket name. This must correspond with the Save Cache to GCS `bucket`.
```

**Review by michaelcretzman — DISMISSED** [review]
Approved with suggestions.

**Pull request merged**